### PR TITLE
Read transcripts

### PR DESCRIPTION
**Pull request by gshellikeri**

### CODE REVIEW

File: frontend/src/PastEncounterList.tsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    let serverFunction = firebase.functions().httpsCallable("getEncounterTranscript");
+    serverFunction({userId: user.uid, encounterId: encounterId}).then(function (response) {
+        console.log("Downloaded transcript to memory");
+        console.log(response.data);
+        let newTranscript = response.data.split ('\n').map((item: React.ReactNode, i: string | number | undefined) => <div key={i}>{item}</div>); 
+        setTranscript(newTranscript);
+    }).catch((err) => {
+        console.log(err);
+    });
+  }, [encounters, transcript, setTranscript]);
```
Comment (tstanis) [inline frontend/src/PastEncounterList.tsx:110]: i don't think you need encounters or transcript here as dependencies.

File: frontend/src/PastEncounterList.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      if (timerHandle) {
+        clearInterval(timerHandle);
+      }
+    };
+  }, [refreshEncounters, refresh]);
+
+  const onView = React.useCallback((event: any, index: number) => {
+    fetchTranscript(encounters[index].encounterId);
+    setViewOpen(true);
+  }, [encounters, setViewOpen, setViewEncounterIndex]);
```
Comment (tstanis) [inline frontend/src/PastEncounterList.tsx]: shouldn't be dependent on setViewEncounterIndex, or at least, you aren't calling it...

File: functions/index.js [hunk 3]
```diff
@@ -216,6 +217,27 @@ exports.listEncounters = functions.https.onRequest((request, response) => {
     });
 });
 
+exports.listEncountersByStatus = functions.https.onRequest((request, response) => {
```
Comment (tstanis) [inline functions/index.js:220]: you might just add this functionality to listEncounters.  Seems like a reasonable option to have to the method in general.

File: functions/index.js [hunk 4]
```diff
@@ -216,6 +217,27 @@ exports.listEncounters = functions.https.onRequest((request, response) => {
...
+    return cors(request, response, () => {
+        var userId = request.body.data.userId;
+        var status = request.body.data.status;
+        let encountersRef = db.collection('encounters').where('state', '==', status);
+        let allEncounters = encountersRef.get()
+            .then(encounters => {
+                var returnEncounters = []
+                encounters.forEach(doc => {
```
Comment (tstanis) [inline functions/index.js:228]: nit: you can use map() here and avoid some code if you want,

File: frontend/src/PastEncounterList.tsx (hunk 1 above)
Comment (gshellikeri) [reply frontend/src/PastEncounterList.tsx]: Yes, fixed.

File: functions/index.js [hunk 5]
```diff
@@ -502,18 +524,20 @@ exports.createTranscript = functions.https.onRequest((request, response) => {
             const metadata = {contentType: "text/html"};
             const bucket = storage.bucket();
             var transcript = request.body.data.transcript;
-            var encounterId = request.body.data.encounterId;
+            var encounter = request.body.data.encounter;
```
Comment (tstanis) [inline functions/index.js:527]: why pass the full encounter rather then just the id?

File: functions/index.js [hunk 6]
```diff
@@ -557,4 +580,25 @@ exports.txtParticipant = functions.https.onRequest((request, response) => {
             response.status(500).send(err);
         });
     });
+});
+
+// Downloads file to memory - revisit this.
+exports.getEncounterTranscript = functions.https.onRequest((request, response) => {
+    return cors(request, response, () => {
+        try {
+            var userId = request.body.data.userId;
```
Comment (tstanis) [inline functions/index.js:589]: is the userId the user for the patient or the clinician?

**Pull request merged**